features/admin.connections.v1/components/edit/settings/connected-apps.tsx

```diff
@@ -152,31 +155,120 @@ export const ConnectedApps: FunctionComponent<ConnectedAppsPropsInterface> = (
 
     const dispatch: Dispatch = useDispatch();
     const { organizationType } = useGetCurrentOrganizationType();
+    const { t } = useTranslation();
 
     const UIConfig: UIConfigInterface = useSelector((state: AppState) => state?.config?.ui);
 
     const [ connectedApps, setConnectedApps ] = useState<ConnectedAppInterface[]>();
     const [ filterSelectedApps, setFilterSelectedApps ] = useState<ConnectedAppInterface[]>([]);
     const [ connectedAppsCount, setconnectedAppsCount ] = useState<number>(0);
-    const [ isAppsLoading, setIsAppsLoading ] = useState<boolean>(true);
+    const [ isCustomLocalAuthenticator, setIsCustomLocalAuthenticator ] = useState<boolean>(undefined);
+    const [ isAppsLoading, setIsAppsLoading ] = useState<boolean>(false);
+    const [ shouldFetchLocalAuthenticatorConnectedApps, setShouldFetchLocalAuthenticatorConnectedApps ] =
+        useState<boolean>(false);
     const [ searchQuery, setSearchQuery ] = useState<string>("");
     const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
     const applicationTemplates: ApplicationTemplateListItemInterface[] = useSelector(
         (state: AppState) => state.application.templates);
     const groupedApplicationTemplates: ApplicationTemplateListItemInterface[] = useSelector(
         (state: AppState) => state?.application?.groupedTemplates);
+
     const [
         isApplicationTemplateRequestLoading,
         setApplicationTemplateRequestLoadingStatus
     ] = useState<boolean>(false);
-
-    const { t } = useTranslation();
     const {
         templates: extensionApplicationTemplates,
         isExtensionTemplatesRequestLoading: isExtensionApplicationTemplatesRequestLoading
     } = useExtensionTemplates();
+    const {
+        data: connectedAppsOfAuthenticator,
+        isLoading: isAuthenticatorConnectedAppsLoading
+    } = useGetAuthenticatorConnectedApps(editingIDP?.id, shouldFetchLocalAuthenticatorConnectedApps);
+
+    /**
+     * This useEffect checks whether the authenticator is a custom local authenticator.
+     * If yes, it enables fetching the connected apps of the local authenticator.
+     */
+    useEffect(() => {
+        /**
+         * Check whether the connection is a custom local authenticator.
+         *
+         * @param idp - Connection to be evaluated.
+         * @returns - whether the connection is a custom local authenticator.
+         */
+        const checkCustomLocalAuthenticator = (idp: ConnectionInterface): boolean => {
```

Comment:
Since this logic is used in authenticator-grid component let's define a util and use it as required.